bin/cylc-cat-log

```diff
@@ -29,227 +29,276 @@ if remrun().execute():
+        action="store_const", const=LIST_MODE_LOCAL, dest="list_mode")
 
     parser.add_option(
         "-y", "--list-remote",
-        help="List a task log directory on the task host.",
-        action="store_true", default=False, dest="list_remote")
+        help="Task job log only: List log directory on the job host",
+        action="store_const", const=LIST_MODE_REMOTE, dest="list_mode")
 
-    (options, args) = parser.parse_args()
-    suite = args[0]
-    suiterc = localdb(options.db).get_suiterc(suite)
-
-    owner = options.db_owner
+    return parser
 
-    # If --host is used, the command re-invokes itself on the remote suite
-    # host. For task logs we have to find retrieve task host from the suite run
-    # db located on the suite host.
 
+def get_suite_log_path(options, suite):
+    """Return file name of a suite log, given the options."""
+    if options.list_mode:
+        basename = "."
+    else:
+        if options.filename:
+            basename = options.filename
+        else:
+            basename = NAME_DEFAULT
+        if options.rotation_num:
+            basename += "." + options.rotation_num
+    return os.path.normpath(os.path.join(
+        GLOBAL_CFG.get_derived_host_item(suite, "suite log directory"),
+        basename))
+
+
+def get_task_job_log_path(
+        options, suite, point, task, submit_num, user_at_host):
+    """Return file name of a task job log, given the options."""
+    if user_at_host and "@" in user_at_host:
+        owner, host = user_at_host.split("@", 1)
+    elif user_at_host:
+        owner, host = (None, user_at_host)
+    else:
+        owner, host = (None, None)
+    if options.list_mode:
+        basename = "."
+    elif options.filename:
+        try:
+            basename = JOB_LOG_DEST_MAP[options.filename]
+        except KeyError:
+            basename = options.filename
+    else:
+        basename = JOB_LOG_DEST_MAP[NAME_DEFAULT]
+    if submit_num != "NN":
+        submit_num = "%02d" % submit_num
+    return os.path.normpath(os.path.join(
+        GLOBAL_CFG.get_derived_host_item(
+            suite, "suite job log directory", host, owner),
+        point, task, submit_num, basename))
+
+
+def get_task_job_attrs(options, suite, point, task, submit_num):
+    """Return (user@host, command0) of a task job log.
+
+    user@host is set if task job is run remotely and for relevant log files.
+    command0 is set if task job if running on a batch system that requires a
+    special command to view stdout/stderr files.
+
+    """
+    if (options.filename in JOB_LOG_LOCAL_ALWAYS or
+             options.list_mode == LIST_MODE_LOCAL):
+        return (None, None)
+    suite_dao = CylcSuiteDAO(
+        os.path.join(
+            GLOBAL_CFG.get_derived_host_item(suite, "suite run directory"),
+            CylcSuiteDAO.DB_FILE_BASE_NAME),
+        is_public=True)
+    task_job_data = suite_dao.select_task_job(None, point, task, submit_num)
```

Comment:
This is not really another connection. The old logic already connects to the run time database. It used to select from the `task_states` table. It now selects from the `task_jobs` table.